libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Notifications.h

```diff
@@ -0,0 +1,119 @@
+//
+//  SFRestAPI+Notifications.h
+//  SalesforceSDKCore
+//
+//  Created by Brianna Birman on 4/7/20.
+//  Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
+// 
+//  Redistribution and use of this software in source and binary forms, with or without modification,
+//  are permitted provided that the following conditions are met:
+//  * Redistributions of source code must retain the above copyright notice, this list of conditions
+//  and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//  conditions and the following disclaimer in the documentation and/or other materials provided
+//  with the distribution.
+//  * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+//  endorse or promote products derived from this software without specific prior written
+//  permission of salesforce.com, inc.
+// 
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+//  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+//  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#import <Foundation/Foundation.h>
+#import "SFRestAPI.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SFRestAPI(Notifications)
+
+/**
+ * Returns a request to fetch the status of notifications, including unread and unseen count.
+ * @param apiVersion API version.
+ */
+- (SFRestRequest *)requestForNotificationsStatus:(NSString *)apiVersion;
+
+/**
+ * Returns a request to fetch the given notification.
+ * @param notificationId ID of notification to fetch.
+ * @param apiVersion API version.
+ */
+- (SFRestRequest *)requestForNotification:(NSString *)notificationId apiVersion:(NSString *)apiVersion;
+@end
+
+NS_SWIFT_NAME(FetchNotificationsRequestBuilder)
+@interface SFSDKFetchNotificationsRequestBuilder: NSObject
+
+/**
+ * Sets the number of notifications to fetch. Max and default are 20.
+ * @param size Number of notifications to fetch.
+ */
+- (SFSDKFetchNotificationsRequestBuilder *)setSize:(NSUInteger)size;
+
+/**
+ * Notifications occurring before the provided date will be fetched. Shouldn't be used with `setAfter`.
+ * @param date ISO 8601 formatted string.
+ */
+- (SFSDKFetchNotificationsRequestBuilder *)setBefore:(NSString *)date;
```

Comment:
The method should take a NSDate object and the builder should be the one in charge of generating the correct formatted string.